canonical/maas-ui
# MAASENG-1173 allow soft stop power mode

## Done

- Add soft stop power mode option to power cycle
- Add soft power off option for IPMI machines

## QA

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit `/MAAS/r/machines` and select one machine from the list of machines
- Click on "Power Cycle" and ensure there is a "Soft power off" option.
- Click on the "Soft power off" and ensure that a side panel shows up.
- Ensure that the side panel has an icon with a tooltip instruction.


- From the list of machines, find a machine with an IPMI power type that is powered on
- From the power column, click on the "Take action" menu and ensure you see the soft off option
- Ensure that the machine turns off  after clicking the "Soft off' option.

## Fixes

Fixes: #2675  .
[MAASENG-1173](https://warthogs.atlassian.net/browse/MAASENG-1173)

## Launchpad issue
lp#1928873

### Design Reference
https://github.com/canonical/maas-ui/issues/2675#issuecomment-1179122505

## Screenshots
![image](https://github.com/canonical/maas-ui/assets/47540149/18c06010-2ba6-40b2-8095-940daeccc97e)
![image](https://github.com/canonical/maas-ui/assets/47540149/7d6d5ef3-2589-4869-9fac-22266537d56a)



[MAASENG-1173]: https://warthogs.atlassian.net/browse/MAASENG-1173?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
